src/c++/perf_analyzer/test_concurrency_manager.cc

```diff
@@ -355,7 +457,231 @@ TEST_CASE("concurrency_free_ctx_ids")
+      params.streaming = true;
+    }
+  }};
+
+  const auto& ParameterizeSequence{[&]() {
+    SUBCASE("non-sequence")
+    {
+      is_sequence_model = false;
+      ParameterizeAsyncAndStreaming();
+    }
+    SUBCASE("sequence")
+    {
+      is_sequence_model = true;
+      params.num_of_sequences = 1;
+      ParameterizeAsyncAndStreaming();
+    }
+  }};
+
+  const auto& ParameterizeMemory{[&]() {
+    SUBCASE("No shared memory")
+    {
+      params.shared_memory_type = NO_SHARED_MEMORY;
+      ParameterizeSequence();
+    }
+    SUBCASE("system shared memory")
+    {
+      params.shared_memory_type = SYSTEM_SHARED_MEMORY;
+      ParameterizeSequence();
+    }
+    SUBCASE("cuda shared memory")
+    {
+      params.shared_memory_type = CUDA_SHARED_MEMORY;
+      ParameterizeSequence();
+    }
+  }};
+
+  ParameterizeMemory();
+
+  TestConcurrencyManager tcm(params, is_sequence_model);
+  std::shared_ptr<MockModelParser> mmp{
+      std::make_shared<MockModelParser>(false, false)};
+  ModelTensor model_tensor{};
+  model_tensor.datatype_ = "INT32";
+  model_tensor.is_optional_ = false;
+  model_tensor.is_shape_tensor_ = false;
+  model_tensor.name_ = "INPUT0";
+  model_tensor.shape_ = {1};
+  mmp->inputs_ = std::make_shared<ModelTensorMap>();
+  (*mmp->inputs_)[model_tensor.name_] = model_tensor;
+
+  std::shared_ptr<MockDataLoader> mdl{std::make_shared<MockDataLoader>()};
+  const std::string json_str{R"(
+  {
+    "data": [
+      {
+        "INPUT0": [2000000000]
+      },
+      {
+        "INPUT0": [2000000001]
+      }
+    ]
+  }
+      )"};
+  mdl->ReadDataFromStr(json_str, mmp->Inputs(), mmp->Outputs());
+
+  std::shared_ptr<ThreadStat> thread_stat{std::make_shared<ThreadStat>()};
+  std::shared_ptr<ConcurrencyWorker::ThreadConfig> thread_config{
+      std::make_shared<ConcurrencyWorker::ThreadConfig>(0)};
+  thread_config->concurrency_ = 1;
+
+  tcm.parser_ = mmp;
+  tcm.data_loader_ = mdl;
+  tcm.using_json_data_ = true;
+  tcm.execute_ = true;
+  tcm.batch_size_ = 1;
+  tcm.max_threads_ = 1;
+  tcm.distribution_ = std::uniform_int_distribution<uint64_t>(
+      0, mdl->GetDataStreamsCount() - 1);
+
+  std::shared_ptr<IWorker> worker{tcm.MakeWorker(thread_stat, thread_config)};
+  std::future<void> infer_future{std::async(&IWorker::Infer, worker)};
+
+  std::this_thread::sleep_for(std::chrono::milliseconds(18));
+
+  early_exit = true;
+  infer_future.get();
+
+  const auto& actual_append_raw_calls{tcm.stats_->num_append_raw_calls};
+  const auto& actual_set_shared_memory_calls{
+      tcm.stats_->num_set_shared_memory_calls};
+
+  if (params.shared_memory_type == NO_SHARED_MEMORY) {
+    CHECK(actual_append_raw_calls > 0);
+    CHECK(actual_set_shared_memory_calls == 0);
+  } else {
+    CHECK(actual_append_raw_calls == 0);
+    CHECK(actual_set_shared_memory_calls > 0);
+  }
+}
+
+/// Check that the using_shared_memory_ is being set correctly
+///
+TEST_CASE("Concurrency - Check setting of InitSharedMemory")
```

Comment:
Isn't this your other story? did you get them both done in one shot?

Reply:
I think that's what his slack message was about, how he combined the two.

Reply:
yea. i managed to get this handled in this PR.
It was a pretty straight forward translation from the request rate side.